packages/core/src/RenderPipeline/BatcherManager.ts

```
@@ -8,19 +8,37 @@ import { SubRenderElement } from "./SubRenderElement";
-  primitiveChunkManagerMask: PrimitiveChunkManager;
+  private _primitiveChunkManager2D: PrimitiveChunkManager;
+  private _primitiveChunkManagerMask: PrimitiveChunkManager;
+  private _primitiveChunkManagerUI: PrimitiveChunkManager;
 
-  constructor(engine: Engine) {
-    this.primitiveChunkManager2D = new PrimitiveChunkManager(engine);
-    this.primitiveChunkManagerMask = new PrimitiveChunkManager(engine, 128);
+  constructor(public engine: Engine) {}
+
+  get primitiveChunkManager2D(): PrimitiveChunkManager {
+    return (this._primitiveChunkManager2D ||= new PrimitiveChunkManager(this.engine));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)